src/main/java/cz/muni/fi/pa165/library/entities/Loan.java

```diff
@@ -4,7 +4,19 @@
 import java.util.Set;
 
 /** @author Martin Páleník 359817 */
-
+/**
+ * This entity populates multiple SingleLoans.
+ *
+ * It implements the following assignment condition:
+ * "Take into account that a person can borrow multiple books in a single loan."
+ *
+ * Although it might be useful for performance or usability
+ * it is not necessary for the functionality of application as
+ * the loan functionality is fully implemented by SingleLoan.
+ *
+ * The class is not persisted intentionally.
```

Comment:
How come this class is not persisted?

Reply:
Because it is redundant. 

Based on the class diagram it should have contained `borrowed:Date`, which would somehow justify its existence. However, the attribute eventually ended up in `SingleLoanRepository` making `SingleLoanRepository` sufficient to provide all the information required by the assignment. The only justification for the use of such a class has already been refuted by the answer from Mr Paces.

> dana pripominka ‘Take into account that a person can borrow multiple books in a single loan’ podle mne jen upozornuje na zachovani kardinality vazby vypujcky a knih. 

This cardinality is already captured in SingleLoanRepository. Moreover, we have been told by Mr Rossi that 3 entities would suffice and there is no point in exceeding the requirements, especially under the time pressure.

> Yes, if you are a team of three, then entities needed will be three - this
> should also **reduce the effort for other requirements** like for creation of
> front-end, as there will be less functionality.

Still, the class might be useful to group `SingleLoan` object on the fly. That is why I have left it there, but decided not to persist it.

Reply:
It is persisted , because it has Entity annotation

Reply:
> This cardinality is already captured in SingleLoanRepository
How?

Reply:
> we have been told by Mr Rossi that 3 entities would suffice

He does not see into our project. We need this entity to provide needed functionality.

Reply:
True. It was all bullshit. Removed.